### ISSUE TITLE
Exit value does not reflect an authentication failure

### ISSUE THREAD

**Issue by darkrho** [scrapinghub/shub#31]
I think if any API call fails (i.e. auth error) then the command should exit with a non-zero status. The example below shows how `shub` exit with zero status regardless of the auth failure.

```
$ shub deploy-egg 1462X       
Building egg in: /home/rolando/foo
zip_safe flag not set; analyzing archive contents...
Deploying dependency to Scrapy Cloud project "1462X"
Deploy failed (403):
{"status": "error", "message": "Authentication failed"}
Deployed eggs list at: https://dash.scrapinghub.com/p/1462X/eggs
$ echo $?
0
```

**Comment by eliasdorneles** [comment]
+1 that error should be handled and give the user a proper actionable message.

**Comment by josericardo** [comment]
This was fixed by #37 and #38
